generic_message.go

```diff
@@ -0,0 +1,55 @@
+package analytics
+
+var _ Message = GenericMessage(nil)
+
+// This type represents any event type sent in a track call as described in
+// https://segment.com/docs/libraries/http/
+type GenericMessage map[string]interface{}
+
+func (msg GenericMessage) internal() {
+	panic(unimplementedError)
+}
```

Comment:
This internal method being part of the interface definition of `Message` is the reason you couldn't declare this type in your own program I believe.

Maybe it's an indicator that we should get rid of it and let users extend the package with their own types wherever they need?

Reply:
Yep, that is the reason, and removing `internal()` is what I was going to propose initially. Maybe that was the right direction? The rationale for _not_ going that way though is that we'd get at least some minimal "standardized" validation logic provided by the library rather than distributing that around to each time a flexible event type is needed. Though it seems this is the first time?

Reply:
Maybe we can solve this too, via an exported validation function, or some other mechanism (e.g. if a custom type claims it's a track, verify that it contains the required fields).

Since `Message` has never been implemented by external types (due to `internal()`), we can still change the interface without breaking programs to gain the features we want.

Reply:
It's right that we can safely do this now, but it would mean opening up the `Message` interface from here on out (for this version at least), and this seems a bit more of a considered change. Is there some further need that sways one way or the other? I find it hard to advocate for a particular direction given a sample size of 1. 😃 

The way I'm breaking this down in my mind is that we have two fundamental needs:

1. A way to send messages that aren't defined by the existing types.
2. Some way to validate that a less restrictive message meets some minimal structural requirements.

We could open up the `Message` interface, which takes care of the first point. But it doesn't enforce the second point. We could provide a hypothetical `Validate(map[string]interface{})` function to do this, but there wouldn't be any way to ensure that the message had been validated by it. If we think that is reasonable, it would fulfill our needs. I just have some initial reservations about the opening the interface.

Reply:
It does seem to go beyond the intent of the PR, so probably better to postpone and keep this discussion as content for the time we choose to tackle this.